From 8ef2be43d3031b1407b0a849920c444fd5072d79 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?O=C4=9Fuzhan=20Yi=C4=9Fit?= <bambu_bambu@windowslive.com>
Date: Wed, 12 Aug 2015 15:53:26 +0300
Subject: [PATCH] policy: Show a simple dialog about optimizing apps

When booting, new apps are optimized with dex2oat. During this process a
progress dialog is displayed (optimizing app 13/74). This dialog has an
indetemninate progress spinner that uses 50% of cpu.

This leaves only 50% for dex2oat, prolonging the time to boot.

Credits to Blefish
---
 .../android/internal/policy/impl/PhoneWindowManager.java | 9 ++++-----
 1 file changed, 4 insertions(+), 5 deletions(-)

diff --git a/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java b/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
index d4950fb2eecf8..baf25f6066657 100644
--- a/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
+++ b/policy/src/com/android/internal/policy/impl/PhoneWindowManager.java
@@ -19,7 +19,7 @@
 import android.app.ActivityManagerNative;
 import android.app.AppOpsManager;
 import android.app.IUiModeManager;
-import android.app.ProgressDialog;
+import android.app.AlertDialog;
 import android.app.SearchManager;
 import android.app.StatusBarManager;
 import android.app.UiModeManager;
@@ -6627,7 +6627,7 @@ public void systemBooted() {
         screenTurningOn(null);
     }
 
-    ProgressDialog mBootMsgDialog = null;
+    AlertDialog mBootMsgDialog = null;
 
     /** {@inheritDoc} */
     @Override
@@ -6646,7 +6646,7 @@ public void showBootMessage(final CharSequence msg, final boolean always) {
                         theme = 0;
                     }
 
-                    mBootMsgDialog = new ProgressDialog(mContext, theme) {
+                    mBootMsgDialog = new AlertDialog(mContext, theme) {
                         // This dialog will consume all events coming in to
                         // it, to avoid it trying to do things too early in boot.
                         @Override public boolean dispatchKeyEvent(KeyEvent event) {
@@ -6674,8 +6674,6 @@ public void showBootMessage(final CharSequence msg, final boolean always) {
                     } else {
                         mBootMsgDialog.setTitle(R.string.android_start_title);
                     }
-                    mBootMsgDialog.setProgressStyle(ProgressDialog.STYLE_SPINNER);
-                    mBootMsgDialog.setIndeterminate(true);
                     mBootMsgDialog.getWindow().setType(
                             WindowManager.LayoutParams.TYPE_BOOT_PROGRESS);
                     mBootMsgDialog.getWindow().addFlags(
@@ -6686,6 +6684,7 @@ public void showBootMessage(final CharSequence msg, final boolean always) {
                     lp.screenOrientation = ActivityInfo.SCREEN_ORIENTATION_NOSENSOR;
                     mBootMsgDialog.getWindow().setAttributes(lp);
                     mBootMsgDialog.setCancelable(false);
+                    mBootMsgDialog.setMessage("");
                     mBootMsgDialog.show();
                 }
                 mBootMsgDialog.setMessage(msg);
